libslide/src/parser.rs

```diff
@@ -0,0 +1,135 @@
+use crate::scanner::types::Token;
+use crate::scanner::types::TokenType;
+
+pub enum Expr {
+    Float(f64),
+    Int(i64),
+    BinOp(BinOp),
+    // I added un op even though I havent implemented + or -
+    UnaryOp(UnaryOp),
+}
+
+pub struct BinOp {
+    pub item: Token,
+    pub lhs: Box<Expr>,
+    pub rhs: Box<Expr>,
+}
+
+pub struct UnaryOp{
+    pub item: Token, 
+    pub rhs: Box<Expr>,
+}
+
+pub struct Parser<'a>{
+    input: &'a Vec<Token>,
+    index: usize,
+    cur_token: Token,
+}
+
+impl <'a> Parser<'a>{
+    pub fn new (input: &Vec<Token>) -> Parser{
+        Parser{
+            input: input.into(),
+            index: 1,
+            cur_token: input[0].clone().into(),
+        }
+    }
+
+    fn get_token(&mut self) -> Token{
+        self.index = self.index+1;
+        return self.input[self.index-1].clone();
```

Comment:
why not return a `&Token`?

Reply:
so we can reassign it. If we don't then we have to borrow it which creates a mess. We can try this as well.